go/grpcweb/trailer.go

```diff
@@ -0,0 +1,26 @@
+//Copyright 2018 Improbable. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package grpcweb
+
+import (
+	"net/http"
+	"strings"
+)
+
+// gRPC-Web spec says that must use lower-case header/trailer names.
+// See "HTTP wire protocols" section in
+// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2
+type trailer struct {
+	http.Header
+}
+
+func (t trailer) Add(key, value string) {
+	lowerKey := strings.ToLower(key)
+
+	// If trailer headers, map to lower these keys.
+	if lowerKey == "trailer" {
+		value = strings.ToLower(value)
+	}
```

Comment:
Is this correct? The require is just for the header key to be lower case, right?

Reply:
That's right. I remove it.